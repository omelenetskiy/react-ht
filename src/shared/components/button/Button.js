import * as React from 'react';
import StyledButton from '../../styled/button';

type Props = {
  size?: string,
  children?: React.Node,
  onClick: SyntheticEvent<>,
};

const Button = ({ children, onClick, size }: Props) => (
  <StyledButton size={size} onClick={onClick}>
    {children}
  </StyledButton>
);

Button.defaultProps = {
  size: 'small',
  children: null,
};

export default Button;
