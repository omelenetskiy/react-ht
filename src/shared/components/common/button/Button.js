import React from "react";

const Button = () => <button className="btn">button</button>;

export default Button;
