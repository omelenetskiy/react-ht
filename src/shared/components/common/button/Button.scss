.button {
  border: 0;
  text-transform: uppercase;
  padding: 7px 15px;
  color: #fff;
  background-color: $danger-color;
  font-weight: bold;
  border-radius: 3px;
  cursor: pointer;
  outline: none;

  &:hover {
    background-color: $danger-color-dark;
  }
}

.buttonOutline {
  background-color: transparent;
  padding: 50px;
}
