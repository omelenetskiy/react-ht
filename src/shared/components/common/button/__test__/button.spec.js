import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import Button from '../Button';

const props = {
  children: 'button',
  onClick: jest.fn(),
  size: 'small',
};

describe('Button', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
