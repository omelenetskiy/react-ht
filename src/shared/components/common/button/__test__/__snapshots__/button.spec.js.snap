// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button___default
  onClick={[MockFunction]}
  size="small"
>
  button
</button___default>
`;
